packages/protocol/contracts/team/proving/ProverSet.sol

```diff
@@ -0,0 +1,71 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.24;
+
+import "@openzeppelin/contracts/interfaces/IERC1271.sol";
+import "@openzeppelin/contracts/utils/cryptography/ECDSA.sol";
+import "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import "../../common/EssentialContract.sol";
+import "../../L1/ITaikoL1.sol";
+
+/// @title ProverSet
+/// @notice A contract that holds TKO token and acts as a Taiko prover. This contract will simply
+/// relay `proveBlock` calls to TaikoL1 so msg.sender doesn't need to hold any TKO.
+/// @custom:security-contact security@taiko.xyz
+contract ProverSet is EssentialContract, IERC1271 {
```

Comment:
I think in order to transfer some TKO on behalf of `this` contract to `TaikoL1`  via `AssignmentHook` we shall have a function something like:

    function approveTaikoL1Contract(
        uint256 amount
    )
        public
        onlyOwner
    {
        IERC20(tko).approve(TAIKO, amount);
    }

Reply:
The init function set allowance for TaikoL1 and the AssignmentHook to the max.

Reply:
Ah yeah, i see now!